frontend/src/stimulus/controllers/dynamic/project.controller.ts

```diff
@@ -32,43 +32,17 @@
 import { Controller } from '@hotwired/stimulus';
 
 export default class ProjectController extends Controller {
-  static targets = [
-    'descriptionToggle',
-    'projectRow',
-    'descriptionRow',
-  ];
-
-  declare readonly descriptionToggleTargets:HTMLAnchorElement[];
-  declare readonly projectRowTargets:HTMLTableRowElement[];
-  declare readonly descriptionRowTargets:HTMLTableRowElement[];
-
-  toggleDescription({ target, params: { projectId } }:{ target:HTMLAnchorElement, params:{ projectId:number } }) {
-    const toggledTrigger = target;
-    const otherTrigger = this.descriptionToggleTargets.find((trigger) => trigger !== toggledTrigger);
-    // The projectId action parameter is automatically typecast to Number
-    // and to compare it with a data attribute it needs to be converted to
-    // a string.
-    const clickedProjectRow = this.projectRowTargets.find((projectRow) => projectRow.getAttribute('data-project-id') === projectId.toString());
-    const projectDescriptionRow = this.descriptionRowTargets.find((descriptionRow) => descriptionRow.getAttribute('data-project-id') === projectId.toString());
-
-    if (clickedProjectRow && projectDescriptionRow) {
-      clickedProjectRow.classList.toggle('-no-highlighting');
-      clickedProjectRow.classList.toggle('-expanded');
-      projectDescriptionRow.classList.toggle('-expanded');
-
-      this.setAriaLive(projectDescriptionRow);
-    }
-
-    if (otherTrigger) {
-      otherTrigger.focus();
-    }
+  connect():void {
+    const longTexts = document.querySelectorAll('.project-long-text');
+    longTexts.forEach((e) => {
+      const isEllipssed= this.isEllipsisActive(e as HTMLElement);
+        if (isEllipssed) {
+          document.querySelectorAll(`[data-show-dialog-id=dialog-${e.id}]`)[0].classList.remove('d-none');
+        }
+    });
```

Comment:
Instead of relying on classes which led me to think about looking for some specific styling you could also use [Targets from Stimulus](https://stimulus.hotwired.dev/reference/targets) for both the text as well as for the button. In combination with moving the controllers domain into the table cell itself, this would make the code for referencing the text and the button very simple.

In other words:
* replace the `ProjectsController` by a controller that is active only within a single table cell, e.g. the project description, status description, ... That means that you would end up with an instance of the controller per cell to be checked.
* have targets on the text as well as on the button.